assets/scripts/passwd-strength-indicator.js

```diff
@@ -0,0 +1,63 @@
+(function (window, document) {
+  const indicator = document.getElementById('password-strength')
+
+  document.getElementById('password').addEventListener('input', function(event) {
+    const strength = getStrength(event.target.value)
+    indicator.value = parseInt(strength.percentage, 10)
+    indicator.setAttribute('class', `pw-${strength.label}`)
+  }, false)
+
+  // Return given password srength as an object {percentage, label}
+  getStrength = function (password) {
+    if (!password && password !== '') {
+      throw new Error('password parameter is missing')
+    }
+    if (!password.length) {
+      return {percentage: 0, label: 'weak'}
+    }
+
+    const charsets = [
+      // upper
+      { regexp: /[A-Z]/g, size: 26 },
+      // lower
+      { regexp: /[a-z]/g, size: 26 },
+      // digit
+      { regexp: /[0-9]/g, size: 10 },
+      // special
+      { regexp: /[!@#$%^&*()_+\-=[\]{};':"\\|,.<>/?]/g, size: 30 }
```

Comment:
To be effective, the strength indicator needs to know the size the characters interval. So "everything except" could not work.

Furthermore, the goal is here to have an idea of the password strength, it is not important if it is not really accurate.

So I agree with you, we could add more character. **But** as this piece of code is duplicated in _onboarding_ and _settings_, we may update both of those apps to be consistent.